Implements JSON parser in pure scopes.

queercat opened the pull request:
# Description

Attempts to resolve #66.

This adds a JSON parser written in scopes, will need to implement solidus / reverse solidus but all the other core features are available. More testing needs to be written and need to write better access methods, @aiverson's implementation uses a [json-match function](https://github.com/Fundament-Software/feathergui/blob/11e13c8c5b93c770d360fbeedbbeb3c323b7c2c5/feather/json.sc#L285) I believe.

# Implementation Overview
See my write-up [here.](https://gist.github.com/queercat/0df53ec2813192fd563924e55eef6a0f)

# Summary
Needs some further review and better testing but core functionality is present.

aiverson reviewed (changes requested):
You broke the entire API and removed every convenience feature while adding a broken parser. It heap allocates tiny objects on the heap multiple times per character of input. The Lexer and parser architecture is very bad. This needs to be rewritten completely.

queercat commented:
> You broke the entire API and removed every convenience feature while adding a broken parser. It heap allocates tiny objects on the heap multiple times per character of input and segfaults on reasonably sized input. The Lexer and parser architecture is very bad. This needs to be rewritten completely.

Firstly I'm sorry, the point of this was not to merge yet I thought I set this as a merge draft, I wanted to get the lexer parser implementation done before I moved on to readding the previously implemented features. I had tested on a variety of inputs but I suppose not enough. I'll look at it again and send some questions. Thank you.
